### PR TITLE
Issue #6586: aligned javadoc/xdoc for RequireThisCheck

### PR DESCRIPTION
**Pull request by pbludov**
Issue #6586 (part of #5750)

No changes in the code.

### CODE REVIEW

File: src/xdocs/config_coding.xml [hunk 1]
```diff
@@ -3843,15 +3852,15 @@ public static class A {
...
-    field1 = f;
+    field = f;
   }
 
-  String addSuffixToField(String field1) {
-    // Overlapping by method argument. Equal to "return field1 = field1 + "suffix";"
-    return field1 += "suffix"; // violation: Reference to instance variable "field1" needs "this".
+  String addSuffixToField(String field) {
+    // Overlapping by method argument. Equal to "return field = field + "suffix";"
+    return field += "suffix"; // violation: Reference to instance variable "field" needs "this".
```
Comment (pbludov) [inline src/xdocs/config_coding.xml]: `field1` renamed to `field` just to fit to the line length limit.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java [hunk 2]
```diff
@@ -1423,7 +1546,7 @@ public FrameType getType() {
     /**
      * A frame initiated on entering a for block; holds local for variable names.
      */
-    public static class ForFrame extends AbstractFrame {
+    private static class ForFrame extends AbstractFrame {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java]: Please move this change to separate commit and issue, we are library so we might not know all how we are used, some might subclass this Check. We need to declare such change.

But I am ok to make them private.

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java]: There is six frame classes inside of `RequireThisCheck`: 
- [MethodFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#L1129)
- [ConstructorFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#L1150)
- [ClassFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#1171)
- [AnonymousClassFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#1359)
- [BlockFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#1384)
- [CatchFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#1405)
- [ForFrame](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#1426)

Olny `CatchFrame` #5351 and `ForFrame` #5392 are `public`.

@rnveach is there any reason to keep `CatchFrame` and `ForFrame` `public`?

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java]: Not to my knowledge.
This may have been a mistake when I added them as I was also looking into creating a utility class like described at https://github.com/checkstyle/checkstyle/issues/5307 and probably copied it from that code when I made the PR. The classes had to be public in the util since they would be used by other classes.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java]: Ok, as it recent code, it is not on purpose.
But still better to move to separate commit, I am ok with minor prefix in commit message.

**Comment by romani** [comment]
This was not merged.
@pbludov , please be accurate with description text, GitHub do autoclose .

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java (hunk 2 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java]: @pbludov , please rebase

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java]: done.
